### ISSUE TITLE
Concept clarification about some implicit creation aspects wanted

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#248]
**(1)**
This is the test content of a parameter configuration file of the TestsuitesManagement:

```
"params" : {
            "global": {
                         ${testdict.key1.key2.key3} : "keyvalue",
```

It must be clear to all that this does not work (the scope '`params.global`' is missing for `testdict`). There is no error while reading the configuration, but `${testdict}` does not exist during test execution.

**(2)**
Crosscheck: This code works (full variable scope '`params.global`' added):

```
"params" : {
            "global": {
                         ${params.global.testdict.key1.key2.key3} : "keyvalue",
```

**(3)**
Having **(1)** in my mind, I would expect that also in next example there is no error while reading the configuration and `${testdict}` does not exist during test execution.

```
"params" : {
            "global": {
                       ${testdict} : {},
```

Instead of this I get the following error:

```
Verification against JSON schema failed: 'Additional properties are not allowed ('testdict' was unexpected)'
          Please put the additional params into 'params': { 'global': {...}
```

Astonishing. `${testdict}` is already defined within `"params.global"`. I would assume here that the parser is able to detect this. Therefore there is no need to check the parameter name `"testdict"` against the top level parameter names.

Seems that there is no strict scope separation within the code of the TestsuitesManagement. In my opinion "`${testdict} : {},`" inside `"params.global"` is not a schema validation issue.

Or - other way round - at least also **(1)** would have to be a schema validation issue.

I am worried about the fact that **(1)** and **(3)** have so much different impacts. And I would prefer to have the behavior aligned.

**Comment by namsonx** [comment]
Hello Holger,

This ticket was discussed and implemented in https://github.com/test-fullautomation/robotframework-testsuitesmanagement/issues/251

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful. Issue can be closed.

**Comment by test-fullautomation** [comment]
Released with RobotFramework AIO 0.12.0
